src/main/kotlin/org/rust/lang/core/psi/RsPsiManager.kt

```diff
@@ -257,19 +255,35 @@ val Project.rustPsiManager: RsPsiManager get() = service()
 val Project.rustStructureModificationTracker: ModificationTracker
     get() = rustPsiManager.rustStructureModificationTracker
 
+/**
+ * Returns [RsPsiManager.rustStructureModificationTracker] if [Crate.origin] == [PackageOrigin.WORKSPACE] or
+ * [RsPsiManager.rustStructureModificationTrackerInDependencies] otherwise
+ */
+val Crate.rustStructureModificationTracker: ModificationTracker
```

Comment:
Maybe allow nullable receiver and use the method in `RsImplItemImplMixin#getCachedImplItem` and `RsTypeAliasImplMixin#getCachedImplItem`

```suggestion
val Crate?.rustStructureModificationTracker: ModificationTracker
```

Reply:
If the receiver is nullable, we can't obtain a `project`